src/hb-aat-layout-feat-table.hh

```diff
@@ -113,17 +121,17 @@ struct FeatureName
   inline hb_aat_layout_feature_type_t get_feature_type () const
   { return (hb_aat_layout_feature_type_t) (unsigned int) feature; }
 
-  inline hb_ot_name_id_t get_feature_name_id () const
-  { return (hb_ot_name_id_t) nameIndex; }
+  inline hb_ot_name_id_t get_feature_name_id () const { return nameIndex; }
 
   inline hb_ot_name_id_t get_feature_setting_name_id (const feat                      *feat,
-						      hb_aat_layout_feature_setting_t  setting) const
+						      hb_aat_layout_feature_setting_t  key) const
   {
-    const UnsizedArrayOf<SettingName>& settings_table = feat+settingTableZ;
-    for (unsigned int i = 0; i < nSettings; i++)
-      if (settings_table[i].get_setting () == setting)
-        return settings_table[i].get_name_id ();
-    return HB_OT_NAME_ID_INVALID;
+    const SettingName* setting = (SettingName*) hb_bsearch (&key, feat+settingTableZ,
+							    nSettings,
+							    SettingName::static_size,
+							    SettingName::cmp);
```

Comment:
Probably good idea to have an UnsizedSortedArrayOf (or SortedUnsizedArrayOf?) and move the logic there. COLR/CPAL needed that IIRC.

Reply:
Yes please.